repose-aggregator/artifacts/docker/build.gradle

```diff
@@ -33,21 +37,89 @@ docker {
 task buildImage(type: DockerBuildImage) {
     finalizedBy 'removeImage'
     inputDir = file("$projectDir/src/docker")
-    tag = "rackerlabs/repose:$reposeVersion"
     buildArgs = ['REPOSE_VERSION': reposeVersion]
     // TODO: Set "forcerm" to true once the Docker Gradle plugin supports it
 }
 
-task pushImage(type: DockerPushImage) {
+task tagImageVersion(type: DockerTagImage) {
+    dependsOn 'buildImage'
+    repository = 'rackerlabs/repose'
+    tag = reposeVersion
+    targetImageId { buildImage.getImageId() }
+}
+
+task tagImageLatest(type: DockerTagImage) {
     dependsOn 'buildImage'
-    imageName = buildImage.getTag()
+    repository = 'rackerlabs/repose'
+    tag = 'latest'
+    targetImageId { buildImage.getImageId() }
+}
+
+task pushImageVersion(type: DockerPushImage) {
+    dependsOn 'tagImageVersion'
+    imageName = 'rackerlabs/repose'
+    tag = reposeVersion
+}
+
+task pushImageLatest(type: DockerPushImage) {
+    dependsOn 'tagImageLatest'
+    imageName = 'rackerlabs/repose'
+    tag = 'latest'
 }
 
 task removeImage(type: DockerRemoveImage) {
     dependsOn 'buildImage'
-    mustRunAfter 'pushImage'
+    mustRunAfter 'tagImageVersion', 'tagImageLatest', 'pushImageVersion', 'pushImageLatest'
     targetImageId { buildImage.getImageId() }
     force = true
 }
 
-project.tasks.getByPath(':release').dependsOn pushImage
+static int compareVersions(String v1, String v2) {
+    List<String> v1Parts = v1.tokenize('.')
+    List<String> v2Parts = v2.tokenize('.')
+
+    int minLen = Math.min(v1Parts.size(), v2Parts.size())
+
+    // Compare each number from the versions until either the numbers do not match,
+    // or the end of a version string is hit.
+    for (int i = 0; i < minLen; ++i) {
```

Comment:
No particular reason. I actually got into the habit of preferring the prefix over the suffix notation when I was writing stuff in C++. If the logic doesn't dictate which to use, the prefix notation is ever-so-slightly more performant. See:
http://stackoverflow.com/a/1813008